libs/domain/cart/entries/src/entries.component.ts

```diff
@@ -1,169 +1,163 @@
-    this.currentlyUpdated$.next(groupKey);
-    this.cartService.updateEntry({
-      groupKey,
-      quantity: e.detail.quantity,
-    });
-  }
+  @asyncState()
+  protected activeEntry = valueType(
+    this.cartController.getEntry(this.removeGroupKey)
+  );
 
-  protected onRemove({ groupKey }: CartEntry): void {
-    this.cartService.deleteEntry({ groupKey });
-  }
+  // TODO: implement loading state
+  protected override render(): TemplateResult | void {
+    if (this.isEmpty) {
+      return this.renderEmpty();
+    }
 
-  protected renderEmpty(): TemplateResult {
     return html`
-      <section>
-        <oryx-icon type="cart"></oryx-icon>
-        <p>Your shopping cart is empty</p>
-        <oryx-button>
-          <button>Shop now</button>
-        </oryx-button>
-      </section>
+      <oryx-heading>
+        <h1>${i18n('cart.totals.<count>-items', { count: 7 })}</h1>
+      </oryx-heading>
+
+      ${repeat(
+        this.entries ?? [],
+        (entry) => entry.groupKey,
+        (entry) =>
+          html`
+            <oryx-cart-entry
+              .key=${entry.groupKey}
+              ?readonly=${this.componentOptions.readonly}
+            ></oryx-cart-entry>
+          `
+      )}
+      ${when(this.removeGroupKey, () =>
+        this.renderConfirmation(this.removeGroupKey)
+      )}
     `;
   }
 
-  protected renderEntries(entriesData: EntriesData): TemplateResult {
-    const { entries, loading, currentlyUpdated, options } = entriesData;
-
-    return html` ${repeat(
-      entries,
-      (entry) => entry.groupKey,
-      (entry) => html`
-        <cart-entry
-          ?inert=${loading}
-          .options=${{
-            ...options,
-            ...entry,
-            disabled: loading,
-            updating: entry.groupKey === currentlyUpdated,
-          }}
-          @submit=${(e: CustomEvent): void => this.onUpdate(e, entry)}
-          @oryx.remove=${(): void => this.onRemove(entry)}
-        ></cart-entry>
-      `
-    )}`;
+  constructor() {
+    super();
+    this.addEventListener('submit', this.onSubmit as EventListener);
   }
 
-  protected renderItemsCount(entriesData: EntriesData): TemplateResult {
-    const { options, entries } = entriesData;
-
-    if (options.hideItemsCount) {
-      return html``;
+  /**
+   * Handles updates on the entry. When the quantity is 0, the entry is going
+   * to be removed unless the component options require to seek for confirmation first.
+   */
+  protected onSubmit(ev: CustomEvent<CartEntryChangeEventDetail>): void {
+    const { groupKey, quantity } = ev.detail;
+
+    if (ev.detail.quantity === 0) {
+      if (this.componentOptions?.silentRemove) {
+        this.removeEntry(ev.detail.groupKey);
+      } else {
+        this.removeGroupKey = ev.detail.groupKey;
+      }
+    } else {
+      this.cartService.updateEntry({ groupKey, quantity }).subscribe(() => {
+        if (this.componentOptions.notifyOnUpdate) {
```

Comment:
I see the as 2 distinct things, especially when we allow for paralle cart entry modifications, it might be convenient to get the confirmation. but removing entries is pretty obvious.